botocore/handlers.py

```diff
@@ -519,6 +519,20 @@ def change_get_to_post(request, **kwargs):
         request.url, request.data = request.url.split('?', 1)
 
 
+def set_list_objects_encoding_type_url(params, **kwargs):
+    if 'EncodingType' not in params:
+        params['EncodingType'] = 'url'
+
+
+def decode_list_object(parsed, **kwargs):
+    # This is needed because we are passing url as the encoding type. Since the
+    # paginator is based on the key, we need to handle it before it can be
+    # round tripped.
+    if 'Contents' in parsed and parsed.get('EncodingType') == 'url':
```

Comment:
Yeah, I think we'd have to add support for that in order to make it backwards compatible.